lib/armadietto.js

```diff
@@ -101,13 +97,13 @@ class Armadietto {
     const method = req.method.toUpperCase();
     const uri = url.parse(req.url, true);
```

Comment:
At some point we should change this code to use url.URL (the WHATWG API), but that doesn't need to be done in this PR.  The Node.js docs say:

> Use of the legacy url.parse() method is discouraged. Users should use the WHATWG URL API. Because the url.parse() method uses a lenient, non-standard algorithm for parsing URL strings, security issues can be introduced. Specifically, issues with host name spoofing and incorrect handling of usernames and passwords have been identified.

Reply:
Yes, certainly. I tried to upgrade this, but `URL` works a bit different from current usage. The problem with Armadietto is that it's using it mostly with (sub)paths, not full URLs, so the URL will be invalid when not also adding a base path/URL as second argument to the `URL` constructor call.

So, I turned this one from a linter error into a warning, to be solved in a separate PR later by anyone who wants to tackle it.